lib/elixir/lib/module/checker.ex

```diff
@@ -1,16 +1,45 @@
 defmodule Module.Checker do
-  def verify(module_map, _binary) do
+  alias Module.ParallelChecker
+
+  @moduledoc false
+
+  def verify(module_map, binary, cache) do
+    warnings = warnings(module_map, cache)
+    binary = chunk(binary, module_map)
+    {binary, warnings}
+  end
+
+  defp chunk(binary, module_map) do
+    checker_chunk = build_chunk(module_map)
+    {:ok, _module, all_chunks} = :beam_lib.all_chunks(binary)
+    {:ok, binary} = :beam_lib.build_module([checker_chunk | all_chunks])
+    binary
+  end
+
+  defp build_chunk(module_map) do
+    map = %{
+      deprecated: :maps.from_list(module_map.deprecated),
+      exports: :maps.from_list(ParallelChecker.definitions_to_exports(module_map.definitions))
+    }
```

Comment:
I was thinking a slightly different representation for the chunks. Because we may add more information in the future, I am thinking the chunk should be in this shape:

    %{
      {name, arity} => {kind, deprecation, inferred}
    }

This means less duplication too.